test/test-v8debugapi.js

```diff
@@ -727,65 +735,139 @@ describe('v8debugapi', function() {
           }));
 
           api.clear(bp);
-          config.capture.maxStringLength = oldMax;
+          config.capture.maxDataSize = oldMaxData;
+          config.capture.maxStringLength = oldMaxLength;
           done();
         });
         process.nextTick(function() {getterObject();});
       });
     });
 
-    it('should limit array length', function(done) {
-      var bp = {
-        id: 'fake-id-124',
-        location: { path: 'test-v8debugapi.js', line: 5 },
-        expressions: ['A']
-      };
-      var oldMax = config.capture.maxProperties;
-      config.capture.maxProperties = 1;
-      api.set(bp, function(err) {
-        assert.ifError(err);
-        api.wait(bp, function(err) {
+    it('should not limit the length of an evaluated array based on maxProperties',
+      function(done) {
+        var bp = {
+          id: 'fake-id-124',
+          location: { path: 'test-v8debugapi.js', line: 5 },
+          expressions: ['A']
+        };
+        var oldMaxProps = config.capture.maxProperties;
+        var oldMaxData = config.capture.maxDataSize;
+        config.capture.maxProperties = 1;
+        config.capture.maxDataSize = 20000;
+        api.set(bp, function(err) {
           assert.ifError(err);
-          var foo = bp.evaluatedExpressions[0];
-          var fooVal = bp.variableTable[foo.varTableIndex];
-          assert.equal(fooVal.members.length, 1);
-          assert(foo.status.description.format.indexOf(
-            'Only first') !== -1);
-          assert(!foo.status.isError);
+          api.wait(bp, function(err) {
+            assert.ifError(err);
+            var foo = bp.evaluatedExpressions[0];
+            var fooVal = bp.variableTable[foo.varTableIndex];
+            if (semver.satisfies(process.version, '<1.6')) {
+              // In v0.12 there are members for the attributes
+              // '1', '2', and '3'
+              assert.equal(fooVal.members.length, 3);
+            }
+            else {
+              // After v0.12 there are members for the attributes 
+              // '1', '2', '3', and 'length'
+              assert.equal(fooVal.members.length, 4);
+            }
+            assert.strictEqual(foo.status, undefined);
 
-          api.clear(bp);
-          config.capture.maxProperties = oldMax;
-          done();
+            api.clear(bp);
+            config.capture.maxDataSize = oldMaxData;
+            config.capture.maxProperties = oldMaxProps;
+            done();
+          });
+          process.nextTick(function() {foo(2);});
```

Comment:
This is a test, so no biggie. If anyone is motivated enough to change things to `setImmediate` it would be better to do it in a follow-on that changes all such in this file.